File: ros/hbba_lite/src/hbba_lite_cpp_tests/test_throttling_hbba_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Int8.h>

#include <hbba_lite/Subscribers.h>

void callback(const std_msgs::Int8::ConstPtr& msg)
{
  ROS_INFO("Data received : %i", static_cast<int>(msg->data));
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "test_on_off_hbba_subscriber");
    ros::NodeHandle nodeHandle;

    ThrottlingHbbaSubscriber<std_msgs::Int8> sub(nodeHandle, "int_topic", 10, &callback);
    ros::spin();

    return 0;
}

File: ros/hbba_lite/scripts/hbba_lite_python_tests/test_throttling_hbba_subscriber.py
#!/usr/bin/env python3
import rospy

from std_msgs.msg import Int8

import hbba_lite


def callback(data):
    rospy.loginfo('Data received : {}'.format(data.data))


def main():
    rospy.init_node('test_throttling_hbba_subscriber')
    sub = hbba_lite.ThrottlingHbbaSubscriber('int_topic', Int8, callback)
    rospy.spin()
    sub.unregister()


if __name__ == '__main__':
    main()
